src/main/java/com/karankumar/bookproject/backend/repository/PredefinedShelfRepository.java

```diff
@@ -19,8 +19,13 @@
 package com.karankumar.bookproject.backend.repository;
 
 import com.karankumar.bookproject.backend.entity.PredefinedShelf;
+import com.karankumar.bookproject.backend.entity.account.User;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.List;
+
 public interface PredefinedShelfRepository extends JpaRepository<PredefinedShelf, Long> {
-    PredefinedShelf findByPredefinedShelfName(PredefinedShelf.ShelfName shelfName);
+    List<PredefinedShelf> findAllByUser(User user);
```

Comment:
Do we really need this method? What would be the use case for this method?

Reply:
this method is used instead of `findAll()`

Reply:
@amit1307 I think this is the same reason for why we need to find all custom shelves for a particular user (we only want to show a user their predefined shelves, not those that belong to other users)